server/src/controllers/internshipModule.ts

```diff
@@ -0,0 +1,93 @@
+import { NextFunction, Request, Response } from "express";
+import { IInternshipModule, InternshipModule } from "../models/internshipModule";
+import { FilterQuery } from "mongoose";
+import { User } from "../models/user";
+import { Forbidden, NotFound } from "http-errors";
+import { Role } from "../authentication/user";
+
+export async function findInternshipModule(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  const user = await User.findOne({ emailAddress: req.user?.email }).populate({
+    path: "studentProfile.internship",
+    populate: { path: "internships" },
+  });
+  if (req.params.id === "my") {
+    if (!user || !user.studentProfile) return next(new NotFound("Student not found"));
+    res.json(user.studentProfile.internship);
+  } else {
+    console.log(
+      "[DEBUG]",
+      req.user?.role,
+      user?.studentProfile?.internship.id,
+      user?.studentProfile?.internship.id !== req.params.id
+    );
+    if (req.user?.role === Role.STUDENT && user?.studentProfile?.internship.id !== req.params.id)
+      return next(new Forbidden("You may only access your own internship module"));
+    const internshipModule = await InternshipModule.findById(req.params.id).populate("internships");
+    if (!internshipModule) return next(new NotFound("Internship module not found"));
+    res.json(internshipModule);
+  }
+}
+
+export async function listInternshipModules(req: Request, res: Response): Promise<void> {
+  const filter: FilterQuery<IInternshipModule> = {};
+  if (req.query.semester) filter.inSemester = req.query.semester as string;
+  res.json(await InternshipModule.find(filter));
+}
+
+export async function listPostponementRequests(req: Request, res: Response): Promise<void> {
+  res.json(
+    await InternshipModule.aggregate([
```

Comment:
ich versteh nicht so richtig was hier passiert? kann man nicht einfach internshipModules nach status POSTPONEMENT_REQUESTED durchsuchen?

Reply:
Ich hab's ein bisschen optimiert, indem jetzt zuerst nach `POSTPONEMENT_REQUESTED` gefiltert wird. Aber der Rest muss leider bleiben, wie er ist (zumindest nach meiner Einschätzung), weil ich ja das aktuellste Event finden muss, und in der Response dann die entsprechenden Fields, die für den Admin relevant sind (`newSemester`, `newSemesterOfStudy`, `reason`) enthalten sein sollen.

Reply:
Macht sinn dass der rest so bleiben muss wegen der events